modules/snooker/src/main/ts/ephox/snooker/api/Generators.ts

```diff
@@ -166,12 +171,55 @@ const merging = function (generators: Generators) {
       // Remove any width calculations because they are no longer relevant.
       Css.remove(raw, 'width');
       Css.remove(cell, 'width');
+
+      Attribute.getOpt(cell, 'scope').each((scope) => Attribute.set(raw, 'scope', scope));
+
       return raw;
     };
   };
 
+  const merge = (cells: SugarElement[]) => {
+    const getScopeProperty = () => {
+      const attributeOpts = Arr.map(cells, (cell) => {
+        const attributeOpt = Attribute.getOpt(cell, 'scope');
+
+        return attributeOpt.map((attribute) => attribute.substr(0, 3));
+      });
+
+      const stringAttributes = Optionals.cat(attributeOpts);
+
+      if (stringAttributes.length === 0) {
+        return Optional.none<string>();
+      } else {
+        const baseScope = stringAttributes[0];
+        const scopes = [ 'row', 'col' ];
+
+        const isMixed = Arr.exists(stringAttributes, (attribute) => {
+          return attribute !== baseScope && Arr.contains(scopes, attribute);
+        });
+
+        return isMixed ? Optional.none<string>() : Optional.from(baseScope);
+      }
+    };
+
+    Css.remove(cells[0], 'width');
+
+    const scope = getScopeProperty();
+
+    if (scope.isSome()) {
```

Comment:
This is a bad place to use `isSome`. You should use `fold` instead, as you're trying to branch based on if this is "some" or "none".